pkg/odo/genericclioptions/runnable.go

```diff
@@ -77,7 +82,12 @@ func GenericRun(o Runnable, cmd *cobra.Command, args []string) {
 		startTelemetry(cmd, err, startTime)
 	}
 	util.LogErrorAndExit(err, "")
-
+	if cmd.Name() == "push" {
```

Comment:
We don't need to filter this just for `odo push` this can be useful thing for all commands.